### PR TITLE
Added OSBuild generation tests

### PR DESCRIPTION
**Pull request by jakub-dzon**
This PR adds tests for OSBuild generation and creation code.

### CODE REVIEW

File: internal/manifests/osbuild_test.go [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+			// when
+			err := creator.Create(ctx, &osBuildConfig)
+
+			//then
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should fail on OSBuildConfig patch failure", func() {
+			// given
+			osBuildConfigRepository.EXPECT().PatchStatus(ctx, gomock.Any(), gomock.Any()).Return(fmt.Errorf("boom"))
```
Comment (bardielle) [inline internal/manifests/osbuild_test.go:151]: I prefer to mention the specific parameter instead of `gomock.Any` (the second parameter - osbuildConfig)

Comment (jakub-dzon) [reply internal/manifests/osbuild_test.go]: Done.

File: internal/manifests/osbuild_test.go [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+					Name:      configName(OSBuildConfigName, 1),
+					Namespace: osBuildConfig.Namespace,
+				},
+				Data: map[string]string{"kickstart": kickstartTxt},
+			}
+		})
+
+		DescribeTable("should create with no kickstart", func(iso *v1alpha1.IsoConfiguration) {
+			// given
+			osBuildConfig.Spec.Template = &v1alpha1.Template{
```
Comment (bardielle) [inline internal/manifests/osbuild_test.go:228]: I think it can be as part of `BeforeEach` function?

**Comment by bardielle** [comment]
@jakub-dzon I had one more comment before merging the UT

**Review by bardielle — APPROVED** [review]

**Pull request merged**